nix/darwin_headerpad_max_install_names.patch

```diff
@@ -0,0 +1,18 @@
+diff --git a/build.zig b/build.zig
+index 85f33dc4d7..d20b6540be 100644
+--- a/build.zig
++++ b/build.zig
+@@ -1325,6 +1325,11 @@
+             .target = resolved_target,
+             .optimize = options.mode,
+         });
++
++        if (resolved_target.result.os.tag == .macos) {
++            shared_lib.headerpad_max_install_names = true;
++        }
++
```

Comment:
I imagine someone would hit this if they tried to package the Tigerbeetle C library in `brew` or Nixpkgs so it should probably go upstream.